pkg/metrics/metrics.go

```diff
@@ -0,0 +1,63 @@
+// Package metrics provides prometheus instrumentation for CAD
+package metrics
+
+import (
+	"os"
+
+	"github.com/openshift/configuration-anomaly-detection/pkg/logging"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/expfmt"
+)
+
+const (
+	namespace      = "cad"
+	investigate    = "investigate"
+	alertTypeLabel = "alert_type"
+	eventType      = "event_type"
+)
+
+// Push collects and pushes metrics to the configured pushgateway
+func Push() {
+	var promPusher *push.Pusher
+	if pushgateway := os.Getenv("CAD_PROMETHEUS_PUSHGATEWAY"); pushgateway != "" {
+		promPusher = push.New(pushgateway, "cad").Format(expfmt.FmtText)
+		promPusher.Collector(Alerts)
+		promPusher.Collector(LimitedSupportSet)
+		promPusher.Collector(LimitedSupportLifted)
+		promPusher.Collector(ServicelogPrepared)
+		err := promPusher.Add()
```

Comment:
Note that there two possible ways to get the metrics to the pushgateway:

- method `Push`
- method `Add`

The difference is that `Push` replace ALL  old metrics with new ones (updates the metrics that are in the new push and deletes those that are not),
whereas `Add` updates the old metrics with new and leaves the unmentioned as they are.


For our example there is no difference, as we are pushing ALL the metrics everytime.

**We should use** `Add` as it saves the previous value in case some metric is not collected or sent properly.